### ISSUE TITLE
ERROR: In <declare-styleable> FontFamilyFont, unable to find attribute android:fontVariationSettings

### ISSUE THREAD

**Issue by rishabhmathur1693** [pwlin/cordova-plugin-file-opener2#197]
I'm getting these two errors while compiling for the Android platform - 

1. ERROR: In <declare-styleable> FontFamilyFont, unable to find attribute android:fontVariationSettings
2. ERROR: In <declare-styleable> FontFamilyFont, unable to find attribute android:ttcIndex

Can anyone please help me figure out why this could be happening?

When I remove this plugin, build process succeeds.

**Comment by rishabhmathur1693** [comment]
Thank you so much @aamatte . You really saved me today. I've spent whole night researching this issue.

**Comment by PRINCEHR** [comment]
plugin.xml can be found in platforms-->android-->project.properties

This will help to find xml file to remove that error

**Comment by nemexis** [comment]
Is not necessary to remove and add the android platform... just rebuild with --clean , after changing the com.android.support:support-v4:+ to com.android.support:support-v4:27.+ . in the project.properties file as described before by aamatte .

**Comment by rdanieli** [comment]
I'd like to inform that to me only worked when I removed and added it again.

**Comment by robindierckx** [comment]
Same issue, but the workaround does not work.

My `platforms-->android-->project.properties` is generated automatically by graddle and looks like this:

```
target=android-26
android.library.reference.1=CordovaLib
cordova.system.library.1=com.android.support:support-v4:24.1.1+
cordova.gradle.include.1=cordova-plugin-crosswalk-webview/creo2storetablet994666-xwalk.gradle
cordova.system.library.2=com.google.android.gms:play-services-analytics:+
cordova.gradle.include.2=cordova-plugin-ignore-lint-translation/creo2storetablet994666-build-extras.gradle

```
As you can see `com.android.support:support-v4:24.1.1+` is asked to be installed (i.e., not v28).

However, after running `ionic cordova build android` I see in a build file `/platforms/android/build/intermediates/blame/res/armv7/debug/multi/values-af.json:` that com.android.support:support-v4:28 seems to be installed:

```
   34                  "from": {
   35:                     "file": "/my-app/platforms/android/build/intermediates/exploded-aar/com.android.support/coordinatorlayout/**28.0.0-alpha1**/res/values/values.xml",
   36                      "position": {
   37                          "startLine": 2,
   ..
  318                  },
```
The issue persist when removing android, and even after completely resetting and reinstalling the whole repo.

Any idea why the specification in project.properties is ignored, and how to avoid installation of com.android.support:support-v4 version 28.0.0-alpha1 ?

**Comment by robindierckx** [comment]
Ok, my problem is solved when uninstalling the crosswalk plugin (that i needed for older devices)

**Comment by richardshergold** [comment]
@robindierckx  I am getting this error in an Ionic project too - what did you do, just change the project.properties file as per the earlier comments?  I don't understand because that file then disappears when you uninstall the android platform and add it again?

**Comment by robindierckx** [comment]
@richardshergold  You are right. The `project.properties` file is generated by `/platforms/android/cordova/lib/builders/GradleBuilder.js`, which is erased when uninstalling the platform. If you don't uninstall the platform though, you can modify your project.properties file searching for com.android.support in GradleBuilder.js, change the version, and then rebuild with --clean.

In my case the version in project.properties was still ok, and the issue came from the fact that for some reason the crosswalk plugin pulled the buggy version of `com.android.support:support-v4`. Removing crosswalk reverted that plugin to an earlier version and removed the bug.

**Comment by RichPC** [comment]
I was getting the same two errors, upgrading phonegap to 8.0.0 solved the problem for me.

**Comment by sreekanth-karini** [comment]
Install this plugin to fix this issue.
ionic cordova plugin add cordova-android-support-gradle-release --save
https://github.com/dpa99c/cordova-android-support-gradle-release

**Comment by hemantdhivar** [comment]
thanks sreekanth, 
this works for me

**Comment by gabrielfreire** [comment]
My problem was **crosswalk-webview plugin**
my project.properties was automatically set to `com.android.support:support-v4:24.1.1+` and it works fine after removing **crosswalk-webview**.

**Comment by gsoulie** [comment]
Still KO for me...2 days ago my project built fine, and since this morning without touch anything, build is KO with this error. crosswalk-webview plugin is not installed in my project, i tried all the solutions mentioned here and nothing works...
I don't understand... Changing to com.android.support:support-v4:24.1.1+ don't do anything

**Comment by appted** [comment]
Same Situation here like **gsoulie** said.

**Comment by coplayer** [comment]
Same fixed error

project.properties file

before
```
cordova.system.library.2=com.android.support:support-v4:+
```
```
cordova.system.library.2=com.android.support:support-v4:27.1.0
```

@aamatte thx

**Comment by jkhax0r** [comment]
I had to go to v4:26.+ for some reason to make it work.  Doing 'cordova platform add android@6.3.0' worked too but seemed like a bigger change

**Comment by ograndebe** [comment]
I'm using crosswalk plugin, and i solved the problem only by using of @sreekanth-karini tip.
running command 
cordova plugin add cordova-android-support-gradle-release --save

I'm not using ionic.

**Comment by williamtran1988** [comment]
cordova plugin add cordova-android-support-gradle-release --save 
it does not help. Issue is still existing.

**Comment by BenjaminKauer** [comment]
For me, it works with cordova-android version 6.3.0.

`ionic cordova platform remove android --save` and `ionic cordova platform add android@6.3.0 --save`.

**Comment by aamirmirza2999** [comment]
when i  import this library 
compile 'com.facebook.android:facebook-core:4.36.1'
    compile 'com.facebook.android:facebook-marketing:4.36.1'
after gradle procces    when i run   found   an error 


ERROR: In <declare-styleable> FontFamilyFont, unable to find attribute android:font
ERROR: In <declare-styleable> FontFamilyFont, unable to find attribute android:fontStyle
ERROR: In <declare-styleable> FontFamilyFont, unable to find attribute android:fontWeight

**Comment by uriklar** [comment]
I was running into this issue with react native and it was caused by the following lines in my `android/app/build.gradle`:

    implementation ("com.google.android.gms:play-services-base:+") {
      force = true;
    }
    implementation ("com.google.android.gms:play-services-maps:+") {
      force = true;
    }

etc...

Apparently this + resolved to a new 17.0.0 version which broke my build.  
Changing the + to 16.0.0 (or 16.0.1 for play-services-base) solved my issue
